### ISSUE TITLE
Discrepancy between flycheck-compile output and Flycheck errors buffer using slim-lint checker

### ISSUE THREAD

**Issue by budu** [flycheck/flycheck#2049]
## Checklist
- [x] I have checked existing issues for potential duplicates before creating this one.
- [x] I have read the [Troubleshooting guide][].

## Bug description
Flycheck with slim-mode is displaying warnings that are not in the `flycheck-compile` output nor can be found in the CLI output.

## Steps to reproduce
Using this slim file:
```
p
  | Can't remember your password for
  strong = @user.email
```

The third line is showing a warning with this text: `Rails/HelperInstanceVariable: Do not use instance variables in helpers. (slim-lint)`. The same warning is listed in the buffer for `flycheck-list-errors`. This warning doesn't make sense in that context.

Running flycheck-compile in the same buffer output:

```
-*- mode: compilation; default-directory: "~/projects/teki/app/views/user_mailer/" -*-
Compilation started at Sun Jan 21 15:49:48

/home/budu/.asdf/shims/slim-lint --reporter\=checkstyle /home/budu/projects/teki/app/views/user_mailer/password_reset.html.slim
<?xml version='1.0'?><checkstyle/>

Compilation finished at Sun Jan 21 15:49:49
```

Running `slim-lint` from the CLI also doesn't report any warnings:

```
$ slim-lint --reporter=checkstyle app/views/user_mailer/password_reset.html.slim
<?xml version='1.0'?><checkstyle/>
```

All paths shown looks correct. The `which` command for `slim-lint` returns the same result everywhere. This executable is the only one in my system.

I've looked at the `flycheck` code and couldn't find anything to explain this behavior.

- [x] I have read https://emacs.stackexchange.com/questions/28429/how-do-i-troubleshoot-emacs-problems

## Expected behavior

There should not be any warnings not found in the `flycheck-compile` output.

## System configuration
```
Syntax checkers for buffer password_reset.html.slim in slim-mode:

First checker to run:

  slim
    - may enable:    yes
    - executable:    Found at /home/budu/.asdf/shims/slimrb
    - next checkers: slim-lint

Checkers that may run as part of the first checker's chain:

  slim-lint
    - may enable: yes
    - executable: Found at /home/budu/.asdf/shims/slim-lint

Flycheck Mode is enabled.  Use C-u C-c ! x to enable disabled checkers.

--------------------

Flycheck version: 33snapshot (package: 20230813.620)
Emacs version:    29.1
System:           x86_64-pc-linux-gnu
Window system:    x

```

Emacs configuration:
- [x] Plain Emacs / Custom configuration
- [ ] Spacemacs
- [ ] Doom Emacs
- [ ] Other shared configuration

My configuration can be found here: https://github.com/budu/my-emacs-config

[Troubleshooting guide]:  https://www.flycheck.org/en/latest/user/troubleshooting.html

**Comment by budu** [comment]
I've been trying to debug this with my very limited Emacs lisp experience and I discovered something very strange. When adding a print call inside of `flycheck-receive-checker-output` like this:

```elisp
(defun flycheck-receive-checker-output (process output)
  "Receive a syntax checking PROCESS OUTPUT."
  (print output)
  (push output (process-get process 'flycheck-pending-output)))
```

The phantom warnings are actually the result of another linter (`rubocop`) which output I can't explain:

```
An error occurred while RSpec/PendingWithoutReason cop was inspecting /tmp/flycheck8iPwdH/password_reset.html.slim.rb:9:29.
***about 20 similarly random line like the above***
An error occurred while RSpec/Pending cop was inspecting /tmp/flycheck8iPwdH/password_reset.html.slim.rb:16:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.60.1 (using Parser 3.3.0.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [x86_64-linux]
<?xml version='1.0'?><checkstyle><file name='/tmp/flycheck8iPwdH/password_reset.html.slim'><error line='6' message='Rails/HelperInstanceVariable: Do not use instance variables in helpers.' severity='warning' source='slim-lint'/><error line='10' message='Rails/HelperInstanceVariable: Do not use instance variables in helpers.' severity='warning' source='slim-lint'/></file></checkstyle>
```

I can't understand why the `rubocop` checker would run here. Calling it directly returns "Cannot use syntax checker ruby-rubocop in this buffer". The error lines in the output doesn't make sense to me. Also why would the `.rb` be appended to the tempory slim file?

**Comment by budu** [comment]
I've found a way to reproduce the issue outside Emacs. The issue arise when linting a file outside the directory of the project. 

So it is a `slim-lint` issue after all.
